### PR TITLE
SWIG fails handling template class with same name as template class in parent namespace

### PR DESCRIPTION
**Pull request by fschlimb**
resolve_create_node_scope treated template classes with same name which it found in a different namespace as a "weak" symbol of the same def. This made it create spurious namespaces and incorrectly reported multiply defined classes.

This is not correct, it can only act so if in the same namespace.
We are now adding a new node in such cases.

fixing issue #807

### CODE REVIEW

**Comment by fschlimb** [comment]
I guess I need some help.

If I am not mistaken, the symtabs indicate "using directives" (e.g. "using namespace X") with the "inherit" key, the "using declaration" (e.g. "using std::pair") with "uname". That's at least what I infer from the failing example. If that's right, functions like ```Swig_symbol_clookup_no_inherit``` use a swapped/inverse terminology in their comments.

Anyway, in ```parser.y:resolve_create_node_scope``` we need to differentiate between a template instantiation and a potentially new template declaration. The test case I added is the latter, but in li_std_pair_using.i it's the former. I do not know how to differentiate.

I need to know the difference, since a new declaration must always create a new node (unless it exists already, but that's an error anyway). However, a specialization/instantiation must not, it must instead use a existing declaration if found. All I have is the name (string), so I cannot decide if I have to create a new node or not.

Any suggestion/idea/help would be appreciated.

**Comment by fschlimb** [comment]
Anyone? @ojwb? @wsfulton?

**Comment by ojwb** [comment]
If it's not obvious from looking at the node in each case, I guess you'll have to add a new flag when such nodes are created so you can tell later.

**Comment by fschlimb** [comment]
Ok, I can investigate this option.
Any hint on where/what "later" could be?

**Comment by ojwb** [comment]
Rereading you want to know in the parser so what I said is probably not useful.  I don't know how you'd tell those apart.

File: Examples/test-suite/same_template_name.i [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+}
+
+namespace ns__a {
+    namespace ns__b {
+
+        template< typename T > class TName : public TName<T> {};
+
+    }
+}
```
Comment (wsfulton) [inline Examples/test-suite/same_template_name.i:15]: This is not valid C++. I used godbolt.org to check and I get compiler errors with up to date compiler versions: gcc-12.2 and  clang-15.0. Interestingly, icc-2021 and msvc-19.3 did not complain with the code above, but an attempt to instantiate the template with the following then resulted in errors:
```C++
void tester() {
  ns__a::ns__b::TName<int> tn;
}
```
Latest SWIG parses it now without error, but instantiating it with %template does lead to incorrect results. I think that is par for the course.

**Comment by wsfulton** [comment]
Closing as invalid.

**Pull request closed without merging**